.gitlab-ci.yml

```diff
@@ -1,463 +1,23 @@
-  <<:                              *kubernetes-build
-  only:
-    - /^[0-9]+$/
-  script:
-    - ./.maintain/gitlab/check_line_width.sh
-  interruptible:                   true
-  allow_failure:                   true
-
-
-
-
-check-polkadot-companion-build:
-  stage:                           build
-  <<:                              *docker-env
-  script:
-    - ./.maintain/gitlab/check_polkadot_companion_build.sh
-  interruptible:                   true
-  allow_failure:                   true
-
-
-cargo-audit:
-  stage:                           test
-  <<:                              *docker-env
-  except:
-    - /^[0-9]+$/
-  script:
-    - cargo audit
-  allow_failure:                   true
-
-cargo-deny:
-  stage:                           test
-  <<:                              *docker-env
-  script:
-    - cargo deny check --hide-inclusion-graph -c .maintain/deny.toml
-  after_script:
-    - echo "___The complete log is in the artifacts___"
-    - cargo deny check -c .maintain/deny.toml 2> deny.log
-  artifacts:
-    name:                          $CI_COMMIT_SHORT_SHA
-    expire_in:                     3 days
-    when:                          always
-    paths:
-      - deny.log
-
-cargo-check-benches:
-  stage:                           test
-  <<:                              *docker-env
-  script:
-    - BUILD_DUMMY_WASM_BINARY=1 time cargo +nightly check --benches --all
-    - sccache -s
-
+  GIT_DEPTH:                       1
 
-cargo-check-subkey:
-  stage:                           test
-  <<:                              *docker-env
-  except:
-    - /^v[0-9]+\.[0-9]+.*$/        # i.e. v1.0, v2.1rc1
-  script:
-    - cd ./bin/utils/subkey
-    - BUILD_DUMMY_WASM_BINARY=1 time cargo check --release
-    - sccache -s
-
-
-test-linux-stable:                 &test-linux
-  stage:                           test
-  <<:                              *docker-env
-  variables:
-    # Enable debug assertions since we are running optimized builds for testing
-    # but still want to have debug assertions.
-    RUSTFLAGS: -Cdebug-assertions=y
-  except:
-    variables:
-      - $DEPLOY_TAG
-  script:
-    - WASM_BUILD_NO_COLOR=1 time cargo test --all --release --verbose --locked |&
-        tee output.log
-    - sccache -s
-    - echo "____Test job successful, checking for warnings____"
-    - awk '/^warning:/,/^$/ { print }' output.log > ${CI_COMMIT_SHORT_SHA}_warnings.log
-    - if [ -s ${CI_COMMIT_SHORT_SHA}_warnings.log ]; then
-        cat ${CI_COMMIT_SHORT_SHA}_warnings.log;
-        exit 1;
-      else
-        echo "___No warnings___";
-      fi
-
-test-dependency-rules:
-  stage:                           test
-  <<:                              *docker-env
-  except:
-    variables:
-      - $DEPLOY_TAG
-  script:
-    - .maintain/ensure-deps.sh
-
-unleash-check:
-  stage:                           test
-  <<:                              *docker-env
+.chaos-docker-release:            &chaos-docker-release
```

Comment:
I would suggest having a general purpose CI job that builds and pushes a docker image for every tested commit. If I remember correctly we can have as many images as we like in public docker hub repositories. The tag of the image would be the shortened git commit hash. (As far as I can tell we only do that for releases right now, correct (//CC @s3krit)?)

The chaostests could then depend on that image.

Reply:
We actually push a docker tag for each successful build of `master` (see https://hub.docker.com/r/parity/substrate/tags ). Currently there are a couple of issues with the `master` build pipeline that I'm working with Ben on, but the general case is that every merge with master gets a docker push. This could potentially change in the future once we settle into a stable release cycle.